sdk/js/src/utils/repairVaa.ts

```
@@ -0,0 +1,137 @@
+            console.log(
+              `found a match for gs index=${currentIndex}, public key=${currentGuardianPublicKey.toString(
+                "hex"
+              )}`
+            );
+            validSignatures.push(signature);
+          } else {
+            console.error(
+              `did not find a match for gs index=${currentIndex}, vaaGuardianPublicKey=${vaaGuardianPublicKey}, current public key=${currentGuardianPublicKey}`
+            );
+          }
+        } catch (err) {
+          console.error(
+            `could not recover address for gs index=${
+              signature.index
+            }, public key=${signature.signature.toString("hex")}`
+          );
+        }
+      });
+
+      // re-construct vaa with signatures that remain
+      const numRepairedSignatures = validSignatures.length;
+      if (numRepairedSignatures < minNumSignatures) {
+        console.error(
+          `There are ${numRepairedSignatures} signatures remaining. Not enough to repair.`
+        );
+        return vaaHex;
+      } else {
+        console.log(
+          `There are ${numRepairedSignatures} signatures remaining. We have enough to repair...\n`
+        );
+        const repairedSignatures = validSignatures
+          .sort(function (a, b) {
+            return a.index - b.index;
+          })
+          .map((signature) => {
+            return `${signature.index
+              .toString(16)
+              .padStart(2, "0")}${signature.signature.toString("hex")}`;
+          })
+          .join("");
+        const newSignatureBody = `${version}${currentGuardianSetIndex
+          .toString(16)
+          .padStart(8, "0")}${numRepairedSignatures
+          .toString(16)
+          .padStart(2, "0")}${repairedSignatures}`;
+
+        const repairedVaa = `${newSignatureBody}${vaaHex.slice(
+          12 + numSignatures * 132
+        )}`;
+        return repairedVaa;
+      }
+    } catch (e) {
```

same as above.
1) the try is too big
2) not sure if worth try-catching anyway. just let the exception propagate